### PR TITLE
VEN-952 | Recurring Invoices

### PR DESCRIPTION
**Pull request by Nizar-Rahme**
## Description :sparkles:
- Implement the recurring invoices page and its subcomponents wrapped in a container 
- Implement a common `DataSummary` component

### Manual testing :construction_worker_man:
- Click on `Kausilaskut` in the sidebar
- The summary section shows how many customers will be invoiced 
- To invoice the customers, click on the `Lähetä venepaikkojen laskut` button which opens a modal
- From the modal select the due date and hit `Lähetä`

## Screenshots :camera_flash:
image

### CODE REVIEW

File: src/features/recurringInvoices/RecurringInvoices.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      </Section>
+      <p>
+        <Trans
+          i18nKey="recurringInvoices.failureInstructions"
+          values={{ failedInvoicesCount: failedInvoicesCount ?? '-' }}
+        >
+          ... <Text as="strong">failedInvoicesCount</Text> ...
+        </Trans>
+      </p>
+      <Table
```
Comment (ahie) [inline src/features/recurringInvoices/RecurringInvoices.tsx:80]: This table might need backend pagination for performance reasons in production, but that can be done in a separate PR later.

File: src/features/recurringInvoices/RecurringInvoices.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+const RecurringInvoices = ({
+  loading,
+  dataSummary,
+  failedInvoicesCount,
+  failedInvoicesData,
+  handleSend,
+}: RecurringInvoicesProps) => {
+  const { t } = useTranslation();
+
+  const failedInvoicesCols: Column<FailedInvoices>[] = [
```
Comment (ahie) [inline src/features/recurringInvoices/RecurringInvoices.tsx:40]: We might actually want a separate column for the date when the failure occurred. But that should be a separate task.

File: src/features/customerView/CustomerViewContainer.tsx [hunk 3]
```diff
@@ -40,7 +40,10 @@ const CustomerViewContainer = () => {
   const [openInvoice, setOpenInvoice] = useState<Invoice>();
   const { t } = useTranslation();
   const { id } = useParams<{ id: string }>();
-  const { loading, data } = useQuery<INDIVIDUAL_CUSTOMER>(INDIVIDUAL_CUSTOMER_QUERY, { variables: { id } });
+  const { loading, data } = useQuery<INDIVIDUAL_CUSTOMER>(INDIVIDUAL_CUSTOMER_QUERY, {
+    variables: { id },
+    fetchPolicy: 'no-cache',
```
Comment (ahie) [inline src/features/customerView/CustomerViewContainer.tsx:45]: Was this query being fetched from the cache previously? I thought we already configured apollo to not use cache? I think there was a PR that fixed some issue by removing lines like this previously.

File: src/common/utils/auth.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import authService from '../../features/auth/authService';
+
+export const getProfileToken = () => {
```
Comment (ahie) [inline src/common/utils/auth.ts:3]: Optional: could this be simply moved to the `AuthService` itself?

**Review by ahie — APPROVED** [review]
Excellent work! Just added a few comments that can be addressed in separate PRs if needed. I'd say :shipit:.

File: src/features/recurringInvoices/RecurringInvoices.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      </Section>
+      <Section>
+        <DataSummary labelValuePairs={dataSummary} />
+      </Section>
+      <p>
+        <Trans
+          i18nKey="recurringInvoices.failureInstructions"
+          values={{ failedInvoicesCount: failedInvoicesCount ?? '-' }}
+        >
+          ... <Text as="strong">failedInvoicesCount</Text> ...
```
[the remark this answers is not in the document]
Comment (Nizar-Rahme) [reply src/features/recurringInvoices/RecurringInvoices.tsx:77]: Exactly, this is just for making the count bold.

File: src/features/customerView/CustomerViewContainer.tsx (hunk 3 above)
Comment (Nizar-Rahme) [reply src/features/customerView/CustomerViewContainer.tsx:45]: This is interesting, It is configured to use `network-only`, but for some reason, Apollo Client was acting weird; when I go to a single customer page, the `useQuery` hook would `return` undefined for the data, even though the actual data is fetched correctly (by observing the requests).
Therefore, this line was needed here to work around the issue.

File: src/features/recurringInvoices/RecurringInvoices.tsx (hunk 2 above)
Comment (Nizar-Rahme) [reply src/features/recurringInvoices/RecurringInvoices.tsx:40]: That makes sense, but there is no such field exposed from the backend, is there?

File: src/features/recurringInvoices/RecurringInvoices.tsx (hunk 1 above)
Comment (Nizar-Rahme) [reply src/features/recurringInvoices/RecurringInvoices.tsx:80]: 👍

File: src/locales/fi.json [hunk 6]
```diff
@@ -598,6 +599,28 @@
...
+    "sendInvoices": "Lähetä venepaikkojen laskut",
+    "failedInvoicesTable": {
+      "header": "Kausilasku ei lähetetty",
+      "name": "Nimi",
+      "harbor": "Satama",
+      "failureReason": "Virheen syy",
+      "emptyState": "Ei laskuja"
+    },
+    "summary": {
+      "customers": "Asiakkaita",
```
[the remark this answers is not in the document]
Comment (Nizar-Rahme) [reply src/locales/fi.json]: Done.

File: src/common/utils/auth.ts (hunk 4 above)
Comment (Nizar-Rahme) [reply src/common/utils/auth.ts:3]: Yup, I will move it in a follow-up PR.

**Pull request merged**